File: num1.swift
import Foundation
func moves(hodi: String, n: Int, m: Int, nlocal: Int, mlocal: Int) -> (Int, Int) {
    var n = n
    var m = m
    
    for hod in hodi {
        switch hod {
        case "L":
            m -= 1
        case "R":
            m += 1
        case "D":
            n += 1
        case "U":
            n -= 1
        default:
            print("\(hod) is undefined")
            return (-123, 0)
        }
        
        if n <= 0 || n > nlocal || m <= 0 || m > mlocal {
            return (0, 0)
        }
    }
    
    return (n, m)
}

func main() {
    var n = 0
    var m = 0
    print("n = ")
    if let input = readLine(), let inputInt = Int(input) {
        n = inputInt
    }
    
    print("m = ")
    if let input = readLine(), let inputInt = Int(input) {
        m = inputInt
    }
    
    var hodi = ""
    print("moves: ")
    if let input = readLine() {
        hodi = input
    }
    
    for i in 1...n {
        for j in 1...m {
            let p = moves(hodi: hodi, n: i, m: j, nlocal: n, mlocal: m)
        
        if (p.0 > 0 && p.0 <= n && p.1 > 0 && p.1 <= m) {
            print(i, j)
            exit(0)
        }
        
        if p.0 == -123 {
            print("No")
            exit(0)
        }
    }
}
    print("No")
}



main()

File: num1.rs
use std::io;

fn moves(hodi: &str, n: i32, m: i32, nlocal: i32, mlocal: i32) -> (i32, i32) {
    let mut n = n;
    let mut m = m;
    let nlocal = nlocal;
    let mlocal = mlocal;
    for hod in hodi.chars() {
        match hod {
            'L' => m -= 1,
            'R' => m += 1,
            'D' => n += 1,
            'U' => n -= 1,
            _ => {
                println!("{} is undefined", hod);
                return (-123, 0);
            }
            
        }
        if m <= 0 || m > mlocal || n <= 0 || n > nlocal {
            return (-123, 0);
        }
    }
    (n, m)
}

fn main() {

    let mut n = String::new();
    let mut m = String::new();
    println!("n = ");
    io::stdin().read_line(&mut n).expect("Failed to read input");
    println!("m = ");
    io::stdin().read_line(&mut m).expect("Failed to read input");
    let n: i32 = n.trim().parse().expect("Invalid input");
    let m: i32 = m.trim().parse().expect("Invalid input");
    
    let mut hodi = String::new();
    io::stdin().read_line(&mut hodi).expect("Failed to read input");
    let hodi = hodi.trim();

    for i in 1..=n {
        for j in 1..=m {
            let p = moves(hodi, i, j, n, m);
            if (p.0 > 0 && p.0 <= n) && (p.1 > 0 && p.1 <= m) {
                println!("{} {}", i, j);
                std::process::exit(0);
            }
            }
        }
    println!("No");
    }
